### PR TITLE
docs: Add Python models docs

### PR DESCRIPTION
**Pull request by chamini2**

### CODE REVIEW

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+sidebar_position: 2
+---
+
+# Making Python dbt models
```
Comment (burkaygur) [inline docsite/docs/Docs/fal-cli/python-models.md:5]: I would reword this to: `Building Python dbt models`

Comment (burkaygur) [inline docsite/docs/Docs/fal-cli/python-models.md:5]: I would reword this to: `Building Python dbt models`

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+> ⚠️ NOTE: This feature is enabled with the flag `--experimental-models`
```
Comment (burkaygur) [inline docsite/docs/Docs/fal-cli/python-models.md:7]: > ⚠️ NOTE: This feature is enabled with the flag `--experimental-models` when running the `fal flow` command.

Comment (burkaygur) [inline docsite/docs/Docs/fal-cli/python-models.md:7]: > ⚠️ NOTE: This feature is enabled with the flag `--experimental-models` when running the `fal flow` command.

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+There are some cases where a SQL transformation is either too complex or impossible. With `fal flow` you can build your dbt models in Python.
```
Comment (burkaygur) [inline docsite/docs/Docs/fal-cli/python-models.md:9]: As of version 0.2.3, fal now supports a new building block: "Pure Python models". This new type of model allows you to represent a dbt model purely in Python code leveraging its rich ecosystem of libraries. With Python models, you can now go beyond the capabilities of SQL and build:

1) Data transformations that Python is better suited for e.g. text manipulation, leveraging external libraries.  
2) Use Data Science libraries such as sklearn, xgboost to represent ML model artifacts such as predictions as dbt models.

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+To achieve this, add the Python (`.py`) or [Notebook (`.ipynb`)](./notebook-files.md) files to any of your [`model-paths`](https://docs.getdbt.com/reference/project-configs/model-paths) and make sure your script calls the [`write_to_model`](../../Reference/variables-and-functions.md#writetomodel-function) function _exactly once_.
+
+## Dependencies
```
Comment (burkaygur) [inline docsite/docs/Docs/fal-cli/python-models.md:13]: `Dependencies on other models`

Comment (burkaygur) [inline docsite/docs/Docs/fal-cli/python-models.md:13]: `Dependencies on other models`

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+`fal` should pick up most usages of [`source`](../../Reference/variables-and-functions.md#source-function) or [`ref`](../../Reference/variables-and-functions.md#ref-function) functions and generate the appropriate dependency for the `dbt` DAG. In case a dependency is not being picked up by `fal`, you can specify them in the top of the Python file in the [module docstring](https://peps.python.org/pep-0257/):
```
Comment (burkaygur) [inline docsite/docs/Docs/fal-cli/python-models.md:15]: `fal` will resolve any usage of [`source`](../../Reference/variables-and-functions.md#source-function) or [`ref`](../../Reference/variables-and-functions.md#ref-function) functions and generate the appropriate dependencies for the `dbt` DAG. In case a dependency is not being picked up by `fal`, you can specify it on top of the Python file in the [module docstring](https://peps.python.org/pep-0257/):

Comment (burkaygur) [inline docsite/docs/Docs/fal-cli/python-models.md:15]: `fal` will resolve any usage of [`source`](../../Reference/variables-and-functions.md#source-function) or [`ref`](../../Reference/variables-and-functions.md#ref-function) functions and generate the appropriate dependencies for the `dbt` DAG. In case a dependency is not being picked up by `fal`, you can specify it on top of the Python file in the [module docstring](https://peps.python.org/pep-0257/):

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+# fal will pick up this dependency
+df_c = ref('model_c')
+
+# calculate dataframe
+df = some_calc(df_c)
+
+write_to_model(df)
+```
+
+## Generated files
```
Comment (burkaygur) [inline docsite/docs/Docs/fal-cli/python-models.md:35]: I would call this `Under the hood`

Comment (burkaygur) [inline docsite/docs/Docs/fal-cli/python-models.md:35]: I would call this `Under the hood`

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+When running `fal flow run --experimental-models`, it will automatically generate a sql file for each Python file found. This is so `dbt` _knows_ about the Python model, which enables using the Python model downstream back in `dbt`.
```
Comment (burkaygur) [inline docsite/docs/Docs/fal-cli/python-models.md:37]: When running `fal flow run --experimental-models`, fal automatically generates a SQL file for each Python model. This is done in order to let `dbt` know about the existence of the Python model. This enables some nice properties such as Python models being available in dbt docs, and the ability to refer to Python models from other dbt models.

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+{{ ref('model_a') }}
+{{ source('database_b', 'table_b') }}
+{{ ref('model_c') }}
+
+*/
+
+SELECT * FROM {{ target.schema }}.{{ model.name }}
+```
+And the `FAL_GENERATED <checksum>` line is to check we are not overwritting any user files.
```
Comment (burkaygur) [inline docsite/docs/Docs/fal-cli/python-models.md:55]: The `FAL_GENERATED <checksum>` line is to make sure that the file is not being modified.

File: docsite/docs/Docs/fal-cli/python-models.md (hunk 6 above)
Comment (chamini2) [reply docsite/docs/Docs/fal-cli/python-models.md:35]: We may even need to add some logs about that...

Comment (burkaygur) [reply docsite/docs/Docs/fal-cli/python-models.md:35]: Sure thing. I was just referring to what the title should read. This is more of an "Advanced Section". We mention that there are generated files in the paragraph itself.

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+---
+
+# Making Python dbt models
+
+> ⚠️ NOTE: This feature is enabled with the flag `--experimental-models`
+
+There are some cases where a SQL transformation is either too complex or impossible. With `fal flow` you can build your dbt models in Python.
+
+To achieve this, add the Python (`.py`) or [Notebook (`.ipynb`)](./notebook-files.md) files to any of your [`model-paths`](https://docs.getdbt.com/reference/project-configs/model-paths) and make sure your script calls the [`write_to_model`](../../Reference/variables-and-functions.md#writetomodel-function) function _exactly once_.
+
```
[the remark this answers is not in the document]
Comment (chamini2) [reply docsite/docs/Docs/fal-cli/python-models.md:17]: What do you think?

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
+---
+sidebar_position: 2
+---
+
+# Creating Python dbt models
+
+> ⚠️ NOTE: This feature is enabled with the flag `--experimental-models` when running the `fal flow run` command.
+
+As of version `0.3.0`, fal now supports a new building block: _Pure Python models_. This new type of model allows you to represent a `dbt` model purely in Python code leveraging its rich ecosystem of libraries. With Python models, you can now go beyond the capabilities of SQL and build:
```
Comment (mederka) [inline docsite/docs/Docs/fal-cli/python-models.md:9]:
```suggestion
As of version `0.3.0`, fal supports a new building block: _Pure Python models_. This type of model allows you to represent Python code as `dbt` model, thereby leveraging its rich ecosystem of libraries. With Python models, you can build:
```

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+1. Data transformations that Python is better suited for e.g. text manipulation, leveraging external libraries.
```
Comment (mederka) [inline docsite/docs/Docs/fal-cli/python-models.md:11]:
```suggestion
1. Data transformations that Python is better suited for, e.g. text manipulation, leveraging external Python modules.
```

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+2. Use Data Science libraries such as sklearn, xgboost to represent ML model artifacts such as predictions as dbt models.
```
Comment (mederka) [inline docsite/docs/Docs/fal-cli/python-models.md:12]:
```suggestion
2. ML models as dbt models by using Data Science libraries such as sklearn and xgboost.
```

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+
+There are some cases where a SQL transformation is either too complex or impossible. With `fal flow` you can build your dbt models in Python.
```
Comment (mederka) [inline docsite/docs/Docs/fal-cli/python-models.md:14]: This paragraph can be part of the first paragraph.

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+# write to data warehouse
+write_to_model(df)
+```
+
+## Dependencies on other models
+
+`fal` will resolve any usage of [`source`](../../Reference/variables-and-functions.md#source-function) or [`ref`](../../Reference/variables-and-functions.md#ref-function) functions and generate the appropriate dependencies for the `dbt` DAG.
+
+Certain complex expressions in the Python AST may not be picked up by `fal`. In that case you can specify the dependencies in the top of the Python as a [module docstring](https://peps.python.org/pep-0257/):
```
Comment (mederka) [inline docsite/docs/Docs/fal-cli/python-models.md:30]:
```suggestion
Certain complex expressions in the Python AST may not be picked up by `fal`. In that case you can specify dependencies in top of the Python as a [module docstring](https://peps.python.org/pep-0257/):
```

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+```py
+"""Generates Python model with forecast data
+
+For fal to pick up these dependencies:
+- ref('model_a')
+- source('database_b', 'table_b')
+"""
+
+from prophet import Prophet
```
Comment (mederka) [inline docsite/docs/Docs/fal-cli/python-models.md:39]: Why are we importing prophet?

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+# calculate dataframe
+df = some_calc(df_c)
+
+# write to data warehouse
+write_to_model(df)
+```
+
+## Under the hood
+
+When running `fal flow run --experimental-models`, `fal` will automatically generate an [ephemeral dbt model](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/materializations/#ephemeral) for each Python model. This is done in order to let `dbt` know about the existence of the Python model. This enables some nice properties such as Python models being available in `dbt` docs, and the ability to refer to (`ref`) Python models from other dbt models.
```
Comment (mederka) [inline docsite/docs/Docs/fal-cli/python-models.md:53]:
```suggestion
When running `fal flow run --experimental-models`, `fal` will automatically generate an [ephemeral dbt model](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/materializations/#ephemeral) for each Python model. This is done in order to let `dbt` know about the existence of a Python model. This enables some nice properties such as Python models being available in `dbt` docs, and the ability to refer to Python models from other dbt models by using `ref`.
```

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+
+> ❗️ NOTE: Generated files should be committed to your repository.
+
+These generated files should not be modified directly by the user. They will look like this:
```
Comment (mederka) [inline docsite/docs/Docs/fal-cli/python-models.md:57]:
```suggestion
These generated files should not be modified directly by the user. They will look somewhat like this:
```

File: docsite/docs/Docs/fal-cli/python-models.md [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+
+{{ ref('model_a') }}
+{{ source('database_b', 'table_b') }}
+{{ ref('model_c') }}
+
+*/
+
+SELECT * FROM {{ target.schema }}.{{ model.name }}
+```
+The `FAL_GENERATED <checksum>` line is to make sure that the file is not being directly modified.
```
Comment (mederka) [inline docsite/docs/Docs/fal-cli/python-models.md:73]:
```suggestion
The `FAL_GENERATED <checksum>` line is there to make sure that the file is not being directly modified.
```

**Review by mederka — COMMENTED** [review]
Great documentation! I just did some minor suggestions. The bigger one is that I think that we should stay away from marketing in docs and keep that in our blog posts.

File: docsite/docs/Docs/fal-cli/python-models.md (hunk 15 above)
Comment (chamini2) [reply docsite/docs/Docs/fal-cli/python-models.md:39]: just an example
